fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java

```diff
@@ -0,0 +1,419 @@
+            } else {
+                nonPartitionConjuncts.add(expr);
+            }
+        }
+    }
+
+    private void computePartitionInfo() throws AnalysisException {
+        long start = System.currentTimeMillis();
+        PartitionPruner partitionPruner = new ListPartitionPruner(columnToPartitionValuesMap, columnToNullPartitions,
+                partitionConjuncts, desc);
+        selectedPartitionIds = partitionPruner.prune();
+        if (selectedPartitionIds == null) {
+            selectedPartitionIds = idToPartitionKey.keySet();
+        }
+        LOG.debug("partition prune cost: {} ms, partitions: {}",
+                (System.currentTimeMillis() - start), selectedPartitionIds);
+
+        noEvalPartitionConjuncts.addAll(((ListPartitionPruner) partitionPruner).getNoEvalConjuncts());
+    }
+
+    public void setSelectedPartitionIds(Collection<Long> selectedPartitionIds) {
+        this.selectedPartitionIds = selectedPartitionIds;
+    }
+
+    public void setIdToPartitionKey(Map<Long, PartitionKey> idToPartitionKey) {
+        this.idToPartitionKey = idToPartitionKey;
+    }
+
+    public List<Expr> getNonPartitionConjuncts() {
+        return nonPartitionConjuncts;
+    }
+
+    public List<Expr> getNoEvalPartitionConjuncts() {
+        return noEvalPartitionConjuncts;
+    }
+
+    public List<Expr> getMinMaxConjuncts() {
+        return minMaxConjuncts;
+    }
+
+    public void setMinMaxTuple(TupleDescriptor tuple) {
+        minMaxTuple = tuple;
+    }
+
+    public void getScanRangeLocations(DescriptorTable descTbl) throws UserException {
+        if (selectedPartitionIds.isEmpty()) {
+            return;
+        }
+
+        long start = System.currentTimeMillis();
+        List<PartitionKey> partitionKeys = Lists.newArrayList();
+        List<DescriptorTable.ReferencedPartitionInfo> partitionInfos = Lists.newArrayList();
+        for (long partitionId : selectedPartitionIds) {
+            PartitionKey partitionKey = idToPartitionKey.get(partitionId);
+            partitionKeys.add(partitionKey);
+            partitionInfos.add(new DescriptorTable.ReferencedPartitionInfo(partitionId, partitionKey));
+        }
+        List<String> partNames = Lists.newArrayListWithCapacity(partitionKeys.size());
+        List<String> partColumnNames = hudiTable.getPartitionColumnNames();
+        for (int i = 0; i < partitionKeys.size(); i++) {
+            descTbl.addReferencedPartitions(hudiTable, partitionInfos.get(i));
+            List<String> partValues = Utils.getPartitionValues(partitionKeys.get(i));
+            if (partValues.size() > 0) {
+                partNames.add(FileUtils.makePartName(partColumnNames, partValues));
+            } else {
+                partNames.add("");
+            }
+        }
+
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiTable.getHudiBasePath()).build();
```

Comment:
I think we should add a HudiUtil to encapsulate these apis

Reply:
I will refactor this later